src/MAKE/OPTIONS/Makefile.kokkos_cuda_mpi

```diff
@@ -55,9 +55,9 @@ MPI_LIB =
 # PATH = path for FFT library
 # LIB = name of FFT library
 
-FFT_INC =    	
+FFT_INC = -DFFT_CUFFT	
```

Comment:
Shouldn't this define be explained in the manual? and added to the CMake configuration scripts?

Reply:
@akohlmey can you point to where I need to add this to the CMake configuration scripts?

Reply:
@akohlmey I grepped for `FFT_MKL` as an example, but I don't see that option anywhere in the CMake configuration scripts.

Reply:
`cmake/Modules/Packages/KSPACE.cmake`
Look for the variable `FFT_VALUES`, which can be `KISS`, `FFTW` or `MKL`.

Reply:
I added the value for CUFFT. I also have an option for OpenMP-threaded FFTW3, that is toggled via the define `-DFFT_FFTW_THREADED`. This is in addition to `-DFFT_FFTW3`. Any ideas on how to add that to CMake?

Reply:
looking at the code, i think the choice of the names for the define is not proper. since they only apply to KOKKOS code, it should be `-DFFT_KOKKOS_CUDA` and `-DFFT_KOKKOS_FFTW_THREADED` (or `-DFFT_CUDA_KOKKOS` and `-DFFT_FFTW_THREADED_KOKKOS`) and those should selected as part of the KOKKOS module, since they don't apply to the normal FFT wrapper (and thus the normal FFT interface may be independent, or - if there are to be any constraints - they would have to be enforced as part of the KOKKOS configuration. Also, this is missing changed to CMake configurations to add the necessary libraries to the linker command line.

Reply:
I'm fine with the name change, but I don't use the CMake build system of LAMMPS for compiling, much less develop for it. So it would be great if you or @junghans could help with this.

Reply:
well, it looks to me, that the documentation for the conventional make process needs some work, too. It is not very clear from the changes i see in the PR, that this is a KOKKOS-only feature.

Reply:
This is not a KOKKOS-only feature when using either FFTW or KISS FFT because the behavior should be exactly the same as when not using Kokkos, i.e. I don't want a different define for those libraries. Kokkos can be turned on and off at runtime, so it should still work if Kokkos isn't being used.

Reply:
> This is not a KOKKOS-only feature when using either FFTW or KISS FFT because the behavior should be exactly the same as when not using Kokkos, i.e. I don't want a different define for those libraries. Kokkos can be turned on and off at runtime, so it should still work if Kokkos isn't being used.

but - as far as i can tell - the threading is only added for the KOKKOS FFT wrapper. also there is no provision of using cuFFT with the regular FFT wrapper (it should fall back to KISSFFT with the unrecognized define). the way i read the changes to the docs, this is not reflected and thus i stand by my claim, that what this PR implements is KOKKOS-only and that should be explained in the docs.

Reply:
What if in the future the GPU package implements cuFFT, or the USER-OMP package implements threaded FFTW, are we going to change the name of the define to not include Kokkos?

Reply:
> What if in the future the GPU package implements cuFFT, or the USER-OMP package implements threaded FFTW, are we going to change the name of the define to not include Kokkos?

a) i suspect, you are misunderstanding me. i am not talking about the *names* of the defines, i am talking about the *documentation* of the defines. the modified docs simply offer -DFFT_CUFFT and -DFFT_FFTW_THREADS as options without any explanation. that is IMO not sufficient and misleading. This is something that would have to be explained at least under the "CMake and make info" header. there it should be made clear, that threaded FFTs are (currently) only available with KOKKOS and when KOKKOS is compiled accordingly (using OpenMP) and then an additional library is require for FFTW. similarly cuFFT is only available with KOKKOS (using CUDA) and then the corresponding library has to be added. also, for non-KOKKOS FFTs, the code would use KISSFFT. 

b) even if those defines *would* have KOKKOS specific names (and i am not saying they should), it would be trivial to turn them on based on a more generic define, if there was at some point additional FFT support for the same feature in other packages or the LAMMPS core code. we've been doing the same thing with `-DFFT_FFTW` for **ages**, originally pointing to FFTW2 and later to FFTW3.

c) we've always maintained, that the documentation should represent the current state of LAMMPS and not what might be in the future. we have to keep in mind, that we have a fast growing group of users that have very little technical expertise, and for those we should make an effort to make the documentation as obvious as possible, even if that feels sometimes redundant and a burden. this is a tall order given the complexity of some parts of LAMMPS, but we should still make an effort. this is even more critical when looking at CMake, as that is the build path aimed more at the less experienced or institutional users (think HPC centers). but to get that one right, we first have to converge as to how to properly describe the conventional build options.